Cubical/Categories/Displayed/Cartesian.agda

```diff
@@ -0,0 +1,308 @@
+                  → ∃![ bᴰ ∈ ob[ b ] ]
+                    Hom[ f ][ aᴰ , bᴰ ])
+
+  -- Discrete opfibrations
+  module _ (isDiscOpfib : isDiscreteOpfibration) where
+    open isDiscreteOpfibration isDiscOpfib
+
+    -- Equality of displayed morphisms is trivial in a discrete opfibration.
+    -- Rectify using isSet-fibers the dependent path on morphisms we get from
+    -- unique-lift.
+    Homᴰ≡-DiscreteOpfibration :
+        {a b : B.ob} {f : B [ a , b ]}
+        {aᴰ : ob[ a ]} {bᴰ : ob[ b ]}
+      → isProp Hom[ f ][ aᴰ , bᴰ ]
+    Homᴰ≡-DiscreteOpfibration {f = f} {aᴰ = aᴰ} {bᴰ = bᴰ} fᴰ f'ᴰ =
+      subst (λ p → PathP ((λ i → Hom[ f ][ aᴰ , p i ])) fᴰ f'ᴰ) (isSet-fibers _ bᴰ bᴰ _ refl) $
+        snd (PathPΣ (isContr→isProp (unique-lift f aᴰ) (bᴰ , fᴰ) (bᴰ , f'ᴰ)))
+
+    -- Any displayed arrow is opcartesian.
+    isOpcartesianDiscreteOpfibration :
+        {a b : B.ob} {f : B [ a , b ]}
+        {aᴰ : ob[ a ]} {bᴰ : ob[ b ]}
+        (fᴰ : Hom[ f ][ aᴰ , bᴰ ])
+      → isOpcartesian fᴰ
+    isOpcartesianDiscreteOpfibration
+      {f = f} {aᴰ = aᴰ} {bᴰ = bᴰ} fᴰ {c = c} {cᴰ = cᴰ} g .equiv-proof compᴰ =
+             -- Rectify lift-g to have the same codomain as compᴰ
+             (subst (λ cᴰ → Hom[ g ][ bᴰ , cᴰ ])
+                    (PathPΣ eq-over-comp .fst)
+                    (lift-g .snd)
+             -- It trivially composes to the expected morphism.
+           , Homᴰ≡-DiscreteOpfibration _ _)
+             -- And it's trivially equal to any other choice.
+         , λ _ → Σ≡Prop (λ _ → isSetHomᴰ _ _) (Homᴰ≡-DiscreteOpfibration _ _)
+         where
+           lift-g : Σ[ cᴰ ∈ ob[ c ] ] Hom[ g ][ bᴰ , cᴰ ]
+           lift-g = unique-lift g bᴰ .fst
+
+           eq-over-comp : (map-snd (fᴰ ⋆ᴰ_) lift-g) ≡ (cᴰ , compᴰ)
+           eq-over-comp = isContr→isProp (unique-lift (f B.⋆ g) aᴰ)
+                                         (map-snd (fᴰ ⋆ᴰ_) lift-g)
+                                         (cᴰ , compᴰ)
+
+    isDiscreteOpfibration→Opcleavage : Opcleavage
+    isDiscreteOpfibration→Opcleavage {a = a} {b = b} f aᴰ =
+      -- Extend the second argument with a proof of opcartesianness.
+      map-snd (λ fᴰ → fᴰ , isOpcartesianDiscreteOpfibration fᴰ)
+              (unique-lift f aᴰ .fst)
+
+    isDiscreteOpfibration→isOpfibration : isOpfibration
+    isDiscreteOpfibration→isOpfibration = ∣ isDiscreteOpfibration→Opcleavage ∣₁
+
+    uniqueOpcleavageDiscreteOpfibration : isContr Opcleavage
+    uniqueOpcleavageDiscreteOpfibration .fst = isDiscreteOpfibration→Opcleavage
+    uniqueOpcleavageDiscreteOpfibration .snd opcleavage i f aᴰ =
+      ΣPathP
+         (pair-paths .fst
+        , ΣPathPProp
+            {B = λ _ f → isOpcartesian f}
+            {u = _ , isOpcartesianDiscreteOpfibration _}
+            {v = snd (opcleavage f aᴰ)}
+            isPropIsOpcartesian (pair-paths .snd))
+        i
+      where
+        pair-paths : _
+        pair-paths = PathPΣ (unique-lift f aᴰ .snd (map-snd fst (opcleavage f aᴰ)))
+
```

Comment:
How about
```suggestion
    uniqueOpcleavageDiscreteOpfibration .snd opcleavage =
      implicitFunExt $ implicitFunExt $ funExt λ f → funExt λ aᴰ →
      ΣPathP $
      map-snd (ΣPathPProp isPropIsOpcartesian) $
      PathPΣ $
      unique-lift f aᴰ .snd (map-snd fst (opcleavage f aᴰ))
```
to avoid filling in the implicits?